src/tasks/Common/Utils.cs

```diff
@@ -214,15 +214,66 @@ public static (int, string) TryRunProcess(
         return (process.ExitCode, outputBuilder.ToString().Trim('\r', '\n'));
     }
 
+    private static bool ContentEqual(string filePathA, string filePathB)
+    {
+        const int bufferSize = 8192;
+        using FileStream streamA = new(filePathA, FileMode.Open, FileAccess.Read, FileShare.Read, bufferSize: bufferSize, FileOptions.SequentialScan);
+        using FileStream streamB = new(filePathB, FileMode.Open, FileAccess.Read, FileShare.Read, bufferSize: bufferSize, FileOptions.SequentialScan);
+
+        if (streamA.Length != streamB.Length)
+            return false;
+
+        byte[] bufferA = new byte[bufferSize];
+        byte[] bufferB = new byte[bufferSize];
+
+        int readA = 0;
+        int readB = 0;
+        int consumedA = 0;
+        int consumedB = 0;
+
+        /*
+            Read both streams in parallel into rolling buffers, comparing overlapping bytes.
+            Advance the consumed amount by the overlap. Refill a buffer when the previous read is exhausted.
+            This keeps the comparison position in sync, even if the read amounts differ.
+         */
+
+        while (true)
+        {
+            if (consumedA == readA)
+            {
+                readA = streamA.Read(bufferA, 0, bufferSize);
+                consumedA = 0;
+            }
+
+            if (consumedB == readB)
+            {
+                readB = streamB.Read(bufferB, 0, bufferSize);
+                consumedB = 0;
+            }
+
+            if (readA == 0 && readB == 0)
+                return true;
+
+            if (readA == 0 || readB == 0)
+                return false;
+
+            int overlap = Math.Min(readA - consumedA, readB - consumedB);
+            if (!bufferA.AsSpan(consumedA, overlap).SequenceEqual(bufferB.AsSpan(consumedB, overlap)))
+                return false;
+
+            consumedA += overlap;
+            consumedB += overlap;
+        }
+    }
+
+#pragma warning disable  IDE0060 // Remove unused parameter
```

Comment:
I'm certain there is a difference between the text comparison and bytewise comparison but I think any value is likely an artifact of history that no longer applies.  I'm happy to remove the setting at the callers and will unless someone objects.